compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -243,6 +258,34 @@ fn compare_method_predicate_entailment<'tcx>(
         ocx.register_obligation(traits::Obligation::new(tcx, cause, param_env, predicate));
     }
 
+    // If we're within a const implementation, we need to make sure that the method
+    // does not assume stronger `~const` bounds than the trait definition.
+    //
+    // This registers the `~const` bounds of the impl method, which we will prove
+    // using the hybrid param-env that we earlier augmented with the const conditions
+    // from the impl header and trait method declaration.
+    if check_const_if_const {
+        for (const_condition, span) in
+            tcx.const_conditions(impl_m.def_id).instantiate_own(tcx, impl_to_placeholder_args)
```

Comment:
shouldn't we instantiate the impl method as identity? the `impl_to_placeholder_args` is used to map the trait, isn't it?

Reply:
no thats trait to placeholder args which maps the trait to impl.

impl to placeholder is precisely identity, but i didnt want to get rid of them here

if u want to review https://github.com/rust-lang/rust/pull/131979, pls do lol